### PR TITLE
feat(source): support ingesting Avro map type to RisingWave map type

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Part of #17743

Previously: (#16948) `FORMAT ... ENCODE AVRO (map.handling.mode = 'jsonb')` will convert Avro map type to RisingWave jsonb type. Other values of `map.handling.mode`, or unspecified will be rejected.

Now `map.handling.mode = 'map'` or unspecified will convert it to the native map type.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

As title and description

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17980?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#17980** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17980?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#17860** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17860?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#17864](https://github.com/risingwavelabs/risingwave/pull/17864) <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17864?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#17690** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17690?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17763** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/17763?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by chenzl25** [comment]
@xxchan is it ready for review right now?

**Comment by xxchan** [comment]
@chenzl25 Yes, it's done. Just rebased

File: src/connector/codec/src/decoder/avro/mod.rs [hunk 1]
```diff
@@ -318,6 +318,29 @@ impl<'a> AvroParseOptions<'a> {
             (DataType::Varchar, Value::Uuid(uuid)) => {
                 uuid.as_hyphenated().to_string().into_boxed_str().into()
             }
+            (DataType::Map(map_type), Value::Map(map)) => {
```
Comment (chenzl25) [inline src/connector/codec/src/decoder/avro/mod.rs:321]: LGTM. BTW do you know why we use apache-avro 0.16.0 instead of 0.17.0 here in this crate?

**Review by chenzl25 — APPROVED** [review]
LGTM, thanks!

File: src/connector/codec/src/decoder/avro/mod.rs (hunk 1 above)
Comment (xxchan) [reply src/connector/codec/src/decoder/avro/mod.rs:321]: - We have a few bugfix patches in our fork https://github.com/risingwavelabs/avro/issues/6
- And 0.17.0 is just released last week https://github.com/risingwavelabs/avro/issues/5

**Pull request merged**